x-pack/plugins/security_solution/public/resolver/store/data/reducer.ts

```diff
@@ -151,12 +154,28 @@ export const dataReducer: Reducer<DataState, ResolverAction> = (state = initialS
           ...state,
           nodeEventsInCategory: action.payload,
         };
+        if (next.nodeEventsInCategory) {
+          next.nodeEventsInCategory.loading = false;
+        }
         return next;
       }
     } else {
       // the action is stale, ignore it
       return state;
     }
+  } else if (action.type === 'appRequestedAdditionalRelatedEvents') {
+    const nextState: DataState = {
+      ...state,
+      nodeEventsInCategory: {
+        nodeID: '',
```

Comment:
I'm not following the logic here.

Reply:
You're spreading in `state.nodeEventsInCategory`. You shouldn't need to add their nullish fields. What TS error are you getting?